devito/mpi/distributed.py

```diff
@@ -42,7 +42,10 @@ def cleanup():
         if init_by_devito and MPI.Is_initialized() and not MPI.Is_finalized():
             MPI.Finalize()
     atexit.register(cleanup)
-except ImportError:
```

Comment:
(@FabioLuporini) maybe instead of carrying that import error around we could just add

```
@classmethod
def Init(cls):
   raise e
```

to the dummy fallback below so that it only triggers if it tries to init MPI.

Reply:
fine be me

Reply:
Note: due to scoping rules, I saved the exception in the class variable.